pkg/windowsdriver/procmon/procmon.go

```diff
@@ -148,23 +140,41 @@ func (wp *WinProcmon) Start() error {
 }
 
 //nolint:revive // TODO(WKIT) Fix revive linter
-func decodeStruct(data []uint8, sz uint32) (t DDProcessNotifyType, pid uint64, imagefile, cmdline string, consumed uint32) {
+func decodeStruct(data []uint8, sz uint32) (start *ProcessStartNotification, stop *ProcessStopNotification, consumed uint32) {
+	start = nil
+	stop = nil
 	if unsafe.Sizeof(DDProcessNotification{}.Size) > uintptr(sz) {
-		return
+		return nil, nil, 0
 	}
 
 	n := *(*DDProcessNotification)(unsafe.Pointer(&data[0]))
 	if n.Size > uint64(sz) {
-		return
+		return nil, nil, 0
 	}
 
 	consumed = uint32(n.Size)
-	pid = uint64(n.ProcessId)
-	t = DDProcessNotifyType(n.NotifyType)
+	t := DDProcessNotifyType(n.NotifyType)
 
 	if t == ProcmonNotifyStart {
-		imagefile = winutil.ConvertWindowsString(data[n.ImageFileOffset : n.ImageFileOffset+n.ImageFileLen])
-		cmdline = winutil.ConvertWindowsString(data[n.CommandLineOffset : n.CommandLineOffset+n.CommandLineLen])
+		imagefile := winutil.ConvertWindowsString(data[n.ImageFileOffset : n.ImageFileOffset+n.ImageFileLen])
+		cmdline := winutil.ConvertWindowsString(data[n.CommandLineOffset : n.CommandLineOffset+n.CommandLineLen])
+		var sidstring string
+		if n.SidLen > 0 {
+			sidstring = winutil.ConvertWindowsString(data[n.SidOffset : n.SidOffset+n.SidLen])
+		}
```

Comment:
For the `imagefile` and `cmdline`, we convert immediately (without a length check) but for the `sidstring`, we check the length first and then decide to convert or not. Is there a reason we for the difference? is it safe to convert `imagefile` and `cmdline` without checking the length? From previous discussions about partially packing the user mode buffer in the driver, just want to make sure that we don't now need to check the length before arbitrarily doing the conversion.

Reply:
we probably should check it; I'll add that to the PR that includes the buffer change.
it will work anyway (because the go slice will have a length zero
`data[n.CommandLineOffset : n.CommandLineOffset+n.CommandLineLen]`
So the right thing will happen with zero length.  But we should be explicit.  will do in the other PR.

Reply:
and in fact when I look at the buffer change PR, that extra check has already been made :)